services/gitlab/gitlab-coverage.tester.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const { isIntegerPercentage } = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('Coverage (branch)')
+  .get('/gitlab-org/gitlab-runner/12-0-stable.json')
+  .expectBadge({
+    label: 'coverage',
+    message: isIntegerPercentage,
+  })
+
+t.create('Coverage (existent branch but coverage not set up)')
+  .get('/gitlab-org/gitlab/master.json')
+  .expectBadge({
+    label: 'coverage',
+    message: 'not set up',
+  })
```

Comment:
This test is failing in the ci check, as the upstream badge does indeed return a coverage percentage.
https://gitlab.com/gitlab-org/gitlab/badges/master/coverage.svg
![](https://gitlab.com/gitlab-org/gitlab/badges/master/coverage.svg)

https://app.circleci.com/pipelines/github/badges/shields/2330/workflows/5e0300af-43b8-445a-a7c8-04bd47cefd2c/jobs/111783

```bash
GitlabCoverage [live] Coverage (existent branch but coverage not set up) [ GET /gitlab-org/gitlab/master.json ] - [ GET /gitlab-org/gitlab/master.json ]
/home/circleci/project/core/service-test-runner/cli.js

AssertionError: message mismatch: expected '70%' to equal 'not set up'
    at Function._expectField (core/service-test-runner/icedfrisby-shields.js:69:51)
    at IcedFrisbyNock.afterJSON.json (core/service-test-runner/icedfrisby-shields.js:54:26)
    at IcedFrisbyNock.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:954:10)
    at invokeNextHook (node_modules/icedfrisby/lib/icedfrisby.js:1003:24)
    at Promise (node_modules/icedfrisby/lib/icedfrisby.js:1017:7)
    at new Promise (<anonymous>)
    at IcedFrisbyNock._runHooks (node_modules/icedfrisby/lib/icedfrisby.js:976:12)
    at IcedFrisbyNock.run (node_modules/icedfrisby/lib/icedfrisby.js:1276:20)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```